external/kinesis-asl/src/test/scala/org/apache/spark/streaming/kinesis/KinesisBackedBlockRDDSuite.scala

```diff
@@ -101,6 +103,36 @@ abstract class KinesisBackedBlockRDDTests(aggregateTestData: Boolean)
     }
   }
 
+  testIfEnabled("Basic reading from Kinesis with modified configurations") {
```

Comment:
I wasn't able to test the actual waiting of Kinesis. I haven't looked at the `PrivateMethodTester ` yet and check how that can help us to test how the vars are picked.
I used this testcase to debug and verify that all the values are passed correctly.